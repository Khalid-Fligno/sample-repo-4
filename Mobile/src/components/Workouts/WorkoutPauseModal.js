import React from "react";
import PropTypes from "prop-types";
import { View, StyleSheet } from "react-native";
import Modal from "react-native-modal";
import Icon from "../../components/Shared/Icon";
import colors from "../../styles/colors";
import CustomBtn from "../Shared/CustomBtn";

const customBtnStyle = {
  borderRadius: 50,
  marginTop: 10,
  backgroundColor: colors.white,
  padding: 20,
};
const customBtnTitleStyle = {
  color: colors.transparentBlackDark,
};

const WorkoutPauseModal = ({
  isVisible,
  handleQuit,
  handleRestart,
  handleSkip,
  handleUnpause,
  exerciseList,
  fitnessLevel,
  currentExerciseIndex,
  reps,
}) => (
  <Modal
    isVisible={isVisible}
    animationIn="fadeIn"
    animationInTiming={800}
    animationOut="fadeOut"
    animationOutTiming={800}
  >
    <View style={{ flex: 1, justifyContent: "center" }}>
      <View style={styles.pauseIconContainer}>
        <Icon name="pause" size={100} color={colors.white} />
      </View>
      <View style={styles.pauseModalContainer}>
        <CustomBtn
          customBtnStyle={customBtnStyle}
          customBtnTitleStyle={customBtnTitleStyle}
          titleCapitalise={true}
          Title="QUIT WORKOUT"
          onPress={handleQuit}
        />
        <CustomBtn
          customBtnStyle={customBtnStyle}
          customBtnTitleStyle={customBtnTitleStyle}
          titleCapitalise={true}
          Title="RESTART THIS SET"
          onPress={() =>
            handleRestart(
              exerciseList,
              fitnessLevel || reps,
              currentExerciseIndex
            )
          }
        />

        {handleSkip && (
          <CustomBtn
            customBtnStyle={customBtnStyle}
            customBtnTitleStyle={customBtnTitleStyle}
            titleCapitalise={true}
            Title="SKIP THIS EXERCISE"
            onPress={() =>
              handleSkip(
                exerciseList,
                fitnessLevel || reps,
                currentExerciseIndex
              )
            }
          />
        )}
        <CustomBtn
          customBtnStyle={customBtnStyle}
          outline={true}
          titleCapitalise={true}
          Title="CONTINUE"
          onPress={handleUnpause}
        />
      </View>
    </View>
  </Modal>
);

WorkoutPauseModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  handleQuit: PropTypes.func.isRequired,
  handleRestart: PropTypes.func.isRequired,
  handleSkip: PropTypes.func,
  handleUnpause: PropTypes.func.isRequired,
  exerciseList: PropTypes.arrayOf(PropTypes.object).isRequired,
  fitnessLevel: PropTypes.string,
  reps: PropTypes.string,
};

WorkoutPauseModal.defaultProps = {
  fitnessLevel: undefined,
  reps: undefined,
  handleSkip: undefined,
};

const styles = StyleSheet.create({
  pauseModalContainer: {
    borderRadius: 4,
    overflow: "hidden",
  },
  pauseIconContainer: {
    alignItems: "center",
    justifyContent: "center",
    paddingBottom: 30,
  },
});

export default WorkoutPauseModal;
