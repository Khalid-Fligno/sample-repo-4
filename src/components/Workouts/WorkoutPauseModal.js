import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import Modal from 'react-native-modal';
import fonts from '../../styles/fonts';
import colors from '../../styles/colors';

export default class WorkoutPauseModal extends React.PureComponent {
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  render() {
    const {
      isVisible,
      handleQuit,
      handleRestart,
      handleUnpause,
      exerciseList,
      fitnessLevel,
      reps,
    } = this.props;
    return (
      <Modal
        isVisible={isVisible}
        animationIn="fadeIn"
        animationInTiming={800}
        animationOut="fadeOut"
        animationOutTiming={800}
        onBackdropPress={handleUnpause}
      >
        <View style={styles.pauseModalContainer}>
          <TouchableOpacity
            onPress={handleQuit}
            style={styles.modalButtonQuit}
          >
            <Text style={styles.modalButtonText}>
              QUIT
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={() => handleRestart(exerciseList, fitnessLevel || reps)}
            style={styles.modalButtonRestart}
          >
            <Text style={styles.modalButtonText}>
              RESTART
            </Text>
          </TouchableOpacity>
          <TouchableOpacity
            onPress={handleUnpause}
            style={styles.modalButtonContinue}
          >
            <Text style={styles.modalButtonText}>
              CONTINUE
            </Text>
          </TouchableOpacity>
        </View>
      </Modal>
    );
  }
}

WorkoutPauseModal.propTypes = {
  isVisible: PropTypes.bool.isRequired,
  handleQuit: PropTypes.func.isRequired,
  handleRestart: PropTypes.func.isRequired,
  handleUnpause: PropTypes.func.isRequired,
  exerciseList: PropTypes.arrayOf(PropTypes.object).isRequired,
  fitnessLevel: PropTypes.string,
  reps: PropTypes.string,
};

WorkoutPauseModal.defaultProps = {
  fitnessLevel: undefined,
  reps: undefined,
};

const styles = StyleSheet.create({
  pauseModalContainer: {
    backgroundColor: colors.white,
    borderRadius: 4,
    overflow: 'hidden',
  },
  modalButtonQuit: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.coral.standard,
    height: 50,
    width: '100%',
    marginBottom: 0,
  },
  modalButtonRestart: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.green.standard,
    height: 50,
    width: '100%',
    marginBottom: 0,
  },
  modalButtonContinue: {
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: colors.charcoal.standard,
    height: 50,
    width: '100%',
    marginBottom: 0,
  },
  modalButtonText: {
    fontFamily: fonts.bold,
    fontSize: 14,
    color: colors.white,
    marginTop: 3,
  },
});
