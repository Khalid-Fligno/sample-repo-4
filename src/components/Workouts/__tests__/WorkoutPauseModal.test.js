import React from 'react';
import renderer from 'react-test-renderer';
import WorkoutPauseModal from '../WorkoutPauseModal';

describe('Workout Pause Modal', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<WorkoutPauseModal
      isVisible
      handleQuit={jest.fn()}
      handleRestart={jest.fn()}
      handleUnpause={jest.fn()}
      exerciseList={[{ test: 'Test' }]}
    />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
