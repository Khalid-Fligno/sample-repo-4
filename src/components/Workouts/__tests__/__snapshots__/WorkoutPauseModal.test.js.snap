// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Workout Pause Modal renders correctly 1`] = `
<Modal
  animationType="none"
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  transparent={true}
  visible={true}
>
  <View
    accessible={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "black",
        "bottom": 0,
        "height": 1334,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
        "width": 750,
      }
    }
  />
  <View
    hideModalContentWhileAnimating={false}
    onModalHide={[Function]}
    onModalWillHide={[Function]}
    onModalWillShow={[Function]}
    pointerEvents="box-none"
    scrollHorizontal={false}
    scrollOffset={0}
    scrollOffsetMax={0}
    scrollTo={null}
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
        "margin": 37.5,
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 0,
          },
        ],
      }
    }
    supportedOrientations={
      Array [
        "portrait",
        "landscape",
      ]
    }
    swipeThreshold={100}
  >
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingBottom": 30,
          }
        }
      >
        <Text />
      </View>
      <View
        style={
          Object {
            "backgroundColor": "rgb(255, 255, 255)",
            "borderRadius": 4,
            "overflow": "hidden",
          }
        }
      >
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(188, 188, 189)",
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 0,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(0, 0, 0)",
                "fontFamily": "GothamBold",
                "fontSize": 14,
                "marginTop": 4,
              }
            }
          >
            QUIT WORKOUT
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(38, 38, 40)",
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 0,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "fontFamily": "GothamBold",
                "fontSize": 14,
                "marginTop": 4,
              }
            }
          >
            RESTART THIS SET
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(238, 48, 66)",
              "height": 50,
              "justifyContent": "center",
              "marginBottom": 0,
              "opacity": 1,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(255, 255, 255)",
                "fontFamily": "GothamBold",
                "fontSize": 14,
                "marginTop": 4,
              }
            }
          >
            CONTINUE
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
